### ISSUE TITLE
Application UI hangs on large files

### ISSUE THREAD

**Issue by andyroidc** [facebookincubator/Keyframes#103]
Hey guys,

Using the Android library here.  When loading large json files, I'm getting around a 2-3 second UI hang.  This was obviously undesired and I narrowed down the issue to being in the deserialization of the asset from the InputStream.  The way I got around the issue was by using an Rx chain and pushing the deserialization off of the UI thread while providing a Looper from a HandlerThread to the Scheduler.  I'm sure this can be achieved in a similar fashion with AsyncTask if needed.

In general, the file still takes around 2 seconds to load while on a background thread.
Our animation files are around 30k - 50k lines long.

**Comment by subtleGradient** [comment]
Yeah, in production we deserialize in the background also. We are also
looking into using flat buffers or something to reduce parsing time and
memory overhead. I'm not sure if there's a GitHub issue open for that yet
